### PR TITLE
KAFKA-3283: Remove beta from new consumer documentation

### PR DESCRIPTION
**Pull request by ijuma**
Include a few clean-ups (also in producer section), mention deprecation plans and reorder so that the new consumer documentation is before the old consumers.

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @hachikuji

File: docs/api.html [hunk 1]
```diff
@@ -38,14 +40,28 @@
...
+<pre>
+	&lt;dependency&gt;
+	    &lt;groupId&gt;org.apache.kafka&lt;/groupId&gt;
+	    &lt;artifactId&gt;kafka-clients&lt;/artifactId&gt;
+	    &lt;version&gt;0.10.0.0&lt;/version&gt;
+	&lt;/dependency&gt;
+</pre>
+
+Examples showing how to use the consumer are given in the
+<a href="http://kafka.apache.org/0100/javadoc/index.html?org/apache/kafka/clients/consumer/KafkaConsumer.html" title="Kafka 0.9.0 Javadoc">javadocs</a>.
```
Comment (hachikuji) [inline docs/api.html]: minor: Do we need to change the title?

Comment (ijuma) [reply docs/api.html]: Good point, fixed.

**Comment by hachikuji** [comment]
LGTM. Thanks for taking care of this!

**Pull request closed without merging**